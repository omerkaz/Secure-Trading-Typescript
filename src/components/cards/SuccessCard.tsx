import { TiInfoLargeOutline } from 'react-icons/ti';
import { Link } from 'react-router-dom';
type Props = {
    header: string;
    children: React.ReactNode;
};
function SuccessCard(props: Props) {
    return (
        <div className="card bg-success text-light" style={{ width: '25rem' }}>
            <TiInfoLargeOutline className="w-100 fs-1" />
            <div className="card-body">
                <h5 className="card-title">{props.header}</h5>
                <p className="card-text">
                    {props.children}
                </p>
                <div className="d-flex justify-content-center">
                    <Link className="card-link text-light" to={'/'}>
                        Ana Sayfaya DÃ¶n
                    </Link>
                </div>
            </div>
        </div>
    );
}

export default SuccessCard;
