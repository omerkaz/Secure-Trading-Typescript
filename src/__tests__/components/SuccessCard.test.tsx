import '@testing-library/jest-dom';
import { render, screen, waitFor } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';

import SuccessCard from '@components/cards/SuccessCard';

describe('SuccessCard elements renders correctly', () => {
    test('snapshot test', () => {
        const componentTree = render(
            <BrowserRouter>
                <SuccessCard
                    header={'TEST HEADER'}
                    children={'TEST CHİLDREN'}
                />
            </BrowserRouter>,
        );
        expect(componentTree).toMatchSnapshot();
    });
    
    test('elements test', () => {
        render(
            <BrowserRouter>
                <SuccessCard
                    header={'TEST HEADER'}
                    children={'TEST CHİLDREN'}
                />
            </BrowserRouter>,
        );
        const header = screen.getByText('TEST HEADER');
        expect(header).toBeInTheDocument();
        const children = screen.getByText('TEST CHİLDREN');
        expect(children).toBeInTheDocument();
    });
});
