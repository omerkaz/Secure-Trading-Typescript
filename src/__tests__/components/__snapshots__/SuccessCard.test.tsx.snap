// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SuccessCard elements renders correctly snapshot test 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="card bg-success text-light"
        style="width: 25rem;"
      >
        <svg
          baseProfile="tiny"
          class="w-100 fs-1"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          version="1.2"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M14.234 16.014l.554-1.104c.808-1.61.897-3.228.253-4.552-.122-.252-.277-.479-.443-.693 1.411-.619 2.402-2.026 2.402-3.664 0-2.206-1.794-4-4-4s-4 1.794-4 4c0 .783.234 1.508.624 2.126-1.696.33-2.806 1.248-2.947 1.375-.716.631-.885 1.68-.405 2.504.324.559.886.909 1.494.98l-.554 1.104c-.808 1.61-.897 3.228-.254 4.552.565 1.164 1.621 1.955 2.972 2.229.413.084.836.127 1.254.127 2.368 0 3.965-1.347 4.14-1.501.716-.63.887-1.678.407-2.503-.325-.556-.887-.909-1.497-.98zm-1.234-12.013c1.104 0 2 .896 2 2s-.896 2-2 2c-1.105 0-2-.896-2-2s.895-2 2-2zm-1.816 14.999c-.271 0-.559-.025-.854-.087-1.642-.334-2.328-1.933-1.328-3.927l1-1.995c.5-.996.47-1.63-.108-2.035-.181-.125-.431-.169-.689-.169-.577 0-1.201.214-1.201.214s1.133-1.001 2.812-1.001c.271 0 .56.025.856.087 1.64.334 2.328 1.933 1.328 3.927l-1 1.993c-.5.998-.472 1.632.106 2.035.181.126.433.169.692.169.577 0 1.2-.212 1.2-.212s-1.133 1.001-2.814 1.001z"
          />
        </svg>
        <div
          class="card-body"
        >
          <h5
            class="card-title"
          >
            TEST HEADER
          </h5>
          <p
            class="card-text"
          >
            TEST CHİLDREN
          </p>
          <div
            class="d-flex justify-content-center"
          >
            <a
              class="card-link text-light"
              href="/"
            >
              Ana Sayfaya Dön
            </a>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="card bg-success text-light"
      style="width: 25rem;"
    >
      <svg
        baseProfile="tiny"
        class="w-100 fs-1"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        version="1.2"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M14.234 16.014l.554-1.104c.808-1.61.897-3.228.253-4.552-.122-.252-.277-.479-.443-.693 1.411-.619 2.402-2.026 2.402-3.664 0-2.206-1.794-4-4-4s-4 1.794-4 4c0 .783.234 1.508.624 2.126-1.696.33-2.806 1.248-2.947 1.375-.716.631-.885 1.68-.405 2.504.324.559.886.909 1.494.98l-.554 1.104c-.808 1.61-.897 3.228-.254 4.552.565 1.164 1.621 1.955 2.972 2.229.413.084.836.127 1.254.127 2.368 0 3.965-1.347 4.14-1.501.716-.63.887-1.678.407-2.503-.325-.556-.887-.909-1.497-.98zm-1.234-12.013c1.104 0 2 .896 2 2s-.896 2-2 2c-1.105 0-2-.896-2-2s.895-2 2-2zm-1.816 14.999c-.271 0-.559-.025-.854-.087-1.642-.334-2.328-1.933-1.328-3.927l1-1.995c.5-.996.47-1.63-.108-2.035-.181-.125-.431-.169-.689-.169-.577 0-1.201.214-1.201.214s1.133-1.001 2.812-1.001c.271 0 .56.025.856.087 1.64.334 2.328 1.933 1.328 3.927l-1 1.993c-.5.998-.472 1.632.106 2.035.181.126.433.169.692.169.577 0 1.2-.212 1.2-.212s-1.133 1.001-2.814 1.001z"
        />
      </svg>
      <div
        class="card-body"
      >
        <h5
          class="card-title"
        >
          TEST HEADER
        </h5>
        <p
          class="card-text"
        >
          TEST CHİLDREN
        </p>
        <div
          class="d-flex justify-content-center"
        >
          <a
            class="card-link text-light"
            href="/"
          >
            Ana Sayfaya Dön
          </a>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
